File: app/src/main/java/com/example/projectexpensetracker/utils/FirebaseConfig.java
package com.example.projectexpensetracker.utils;
import com.example.projectexpensetracker.models.*;
import com.example.projectexpensetracker.database.*;

/**
 * FirebaseConfig — Centralized configuration class for Firebase.
 * You should paste your Database URL here.
 */
public class FirebaseConfig {

    // This is your "Environment Variable". 
    // Please replace this with your Realtime Database URL from the Firebase Console.
    public static final String DATABASE_URL = "https://expenseprojecttracker-default-rtdb.asia-southeast1.firebasedatabase.app/";

    // This logic helps determine the data storage node for the user.
    public static String getUserNode(String username) {
        return "users/" + username.replace(".", "_");
    }

    // Logic for determining the account storage node.
    public static String getAccountNode(String username) {
        return "accounts/" + username.replace(".", "_");
    }
}

File: UserApp/firebaseConfig.js
import { initializeApp } from 'firebase/app';
import { getDatabase } from 'firebase/database';

const firebaseConfig = {
  databaseURL: "https://expenseprojecttracker-default-rtdb.asia-southeast1.firebasedatabase.app/",
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);

// Initialize Realtime Database and get a reference to the service
export const database = getDatabase(app);
